dspace-xmlui/src/main/resources/aspects/Submission/submission.js

```diff
@@ -248,6 +248,16 @@ function submissionControl(collectionHandle, workspaceID, initStepAndPage)
 {
 	//load initial submission information
 	var submissionInfo = getSubmissionInfo(workspaceID);
+	
+    if(!collectionHandle.equals(submissionInfo.getCollectionHandle())){
+        // Handle selected does not match the handle of the submission (this can
+        // happen if user has used the back button to change collection). Create
+        // a new workspace item. 
+        var dso = HandleManager.resolveToObject(getDSContext(), collectionHandle);
+        var workspace = WorkspaceItem.create(getDSContext(), dso, true);
```

Comment:
As noted above, this looks like it'd create a new WorkspaceItem each time you hit the backbutton. May need rework.